presto-hive/src/main/java/com/facebook/presto/hive/pagefile/PageDataSink.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.pagefile;
+
+import com.facebook.presto.spi.Page;
+
+import java.io.IOException;
+
+public interface PageDataSink
+{
+    /**
+     * Number of bytes written to this sink so far.
+     */
+    long size();
+
+    /**
+     * Gets the size of the memory buffers.
+     */
+    long getRetainedSizeInBytes();
+
+    /**
+     * Write a page
+     */
+    void write(Page dataPage)
```

Comment:
From what I understand this interface is added by analogy to the `OrcDataSink`.

The main difference that I see right now is that the `OrcDataSink` accepts the data for the entire block in a single call. In other words the buffering of the block is happening before calling the `OrcDataSink#write`. The buffering happens in the OrcWriter. 

From one side it leads to somehow tricky and not very obvious "side effects" or implications of the ORC stripe size. By changing the stripe size we are effectively changing the block size of the underlying file system. Thus in some sense it feels like this logic must be pushed down to the `PageDataSink` interface, so the buffering and block size configuration happens there. But then I realize that if we want to add the meta-information to make the format splittable, we don't wanna add check points in a middle of a block. And if the block buffering is hidden behind the `PageDataSink` interface there will be no way to know where are the block breaking point.

Based on this thoughts it feels like we should stick with the existing ORC model, when the entire block is supplied. We can make this method to accept a `List<Page>`.

Reply:
Changed it to `void write(Iterable<SerializedPage> pages)` ,   the remaining work is to write footer.  After some deeper investigation,  `OrcDataOutput` is mainly designed for multiple streams from all column writers,  PageFile would have only one `outputstream`,  l plan to wrap footer into a slice and not mix it with `SerializedPage`.